File: Classes/EVEOnlineAPI/Corporation/EVEMemberSecurity.m
//
//  EVEMemberSecurity.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 7/26/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEMemberSecurity.h"

@implementation EVEMemberSecurityMember

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"characterID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"name":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"roles":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"grantableRoles":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"rolesAtHQ":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"grantableRolesAtHQ":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"rolesAtBase":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"grantableRolesAtBase":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"rolesAtOther":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"grantableRolesAtOther":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityRoleItem class]},
				   @"titles":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityTitleItem class]}};
	
	return scheme;
}

@end


@implementation EVEMemberSecurityRoleItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"roleID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"roleName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	
	return scheme;
}

@end


@implementation EVEMemberSecurityTitleItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"titleID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"titleName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	
	return scheme;
}


@end


@implementation EVEMemberSecurity

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"members":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMemberSecurityMember class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEMemberSecurity.swift
//
//  EVEMemberSecurity.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEMemberSecurityRole: EVEObject {
	public var roleID: Int64 = 0
	public var roleName: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"roleID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
			"roleName":EVESchemeElementType.String(elementName:nil, transformer:nil),
		]
	}
}

public class EVEMemberSecurityTitle: EVEObject {
	public var titleID: Int = 0
	public var titleName: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"titleID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"titleName":EVESchemeElementType.String(elementName:nil, transformer:nil),
		]
	}
}

public class EVEMemberSecurityMember: EVEObject {
	public var characterID: Int = 0
	public var name: String = ""
	public var roles: [EVEMemberSecurityRole] = []
	public var grantableRoles: [EVEMemberSecurityRole] = []
	public var rolesAtHQ: [EVEMemberSecurityRole] = []
	public var grantableRolesAtHQ: [EVEMemberSecurityRole] = []
	public var rolesAtBase: [EVEMemberSecurityRole] = []
	public var grantableRolesAtBase: [EVEMemberSecurityRole] = []
	public var rolesAtOther: [EVEMemberSecurityRole] = []
	public var grantableRolesAtOther: [EVEMemberSecurityRole] = []
	public var titles: [EVEMemberSecurityTitle] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
		"characterID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		"name":EVESchemeElementType.String(elementName:nil, transformer:nil),
		"roles":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"grantableRoles":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"rolesAtHQ":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"grantableRolesAtHQ":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"rolesAtBase":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"grantableRolesAtBase":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"rolesAtOther":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"grantableRolesAtOther":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityRole.self, transformer: nil),
		"titles":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityTitle.self, transformer: nil),
		]
	}
}


public class EVEMemberSecurity: EVEResult {
	public var members: [EVEMemberSecurityMember] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"members":EVESchemeElementType.Rowset(elementName: nil, type: EVEMemberSecurityMember.self, transformer: nil),
		]
	}
}
